src/Raven.Studio/typescript/viewmodels/resources/widgets/storageWidget.ts

```diff
@@ -109,6 +110,39 @@ class storageWidget extends websocketBasedWidget<Raven.Server.Dashboard.Cluster.
         if (stats) {
             action(stats);
         }
+
+        this.scaleDriveSize();
+    }
+
+    openWidgetSettings(): void {
+        super.openWidgetSettings();
+
+        const openSettingsDialog = new storageWidgetSettings();
+        app.showBootstrapDialog(openSettingsDialog).done(() => this.scaleDriveSize());
+    }
+    
+    private scaleDriveSize(): void {
+        
+        console.log('in scale');
+        
+        const scaleToSize: boolean = localStorage.getObject(storageWidgetSettings.localStorageName);
+        
+        let biggestSize = 0;
+        for (const nodeStat of this.nodeStats()) {
+            for (const mountPoint of nodeStat.mountPoints()) {
+                const size = mountPoint.totalCapacity();
+                if (size > biggestSize) {
+                    biggestSize = size;
+                }
+            }
+        }
+
+        for (const nodeStat of this.nodeStats()) {
+            for (const mountPoint of nodeStat.mountPoints()) {
+                const scaleFactor = scaleToSize ? mountPoint.totalCapacity() / biggestSize * 100 : 100;
```

Comment:
can we avoid iterate on this twice?

Reply:
but we need to calculate `biggestSize` first

Reply:
ok right